airflow/providers/salesforce/hooks/tableau.py

```diff
@@ -0,0 +1,110 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from enum import Enum
+from typing import Optional
+
+from tableauserverclient import Pager, PersonalAccessTokenAuth, Server, TableauAuth
+from tableauserverclient.server import Auth
+
+from airflow.hooks.base_hook import BaseHook
+
+
+class TableauJobFinishCode(Enum):
+    """
+    The finish code indicates the status of the job.
+
+    .. seealso:: https://help.tableau.com/current/api/rest_api/en-us/REST/rest_api_ref.htm#query_job
+
+    """
+    PENDING = -1
+    SUCCESS = 0
+    ERROR = 1
+    CANCELED = 2
+
+
+class TableauHook(BaseHook):
+    """
+    Connects to the Tableau Server Instance and allows to communicate with it.
+
+    .. seealso:: https://tableau.github.io/server-client-python/docs/
+
+    :param site_id: The id of the site where the workbook belongs to.
+        It will connect to the default site if you don't provide an id.
+    :type site_id: Optional[str]
+    :param tableau_conn_id: The Tableau Connection id containing the credentials
+        to authenticate to the Tableau Server.
+    :type tableau_conn_id: str
+    """
+
+    def __init__(self, site_id: Optional[str] = None, tableau_conn_id: str = 'tableau_default'):
```

Comment:
I asked in previous review about the switching the default of site_id from empty string to None. You had a good point that empty string is the default of the pacakge
https://tableau.github.io/server-client-python/docs/api-ref#authentication
Why did you enetualy decided to switch to None?

Reply:
https://github.com/apache/airflow/pull/7410/files#diff-6332846625a8d68be6a4f2b74a47cc5dR56 takes care of that !